tests/array_list_with_map_test.c

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/array_list_with_map.h>
+#include <aws/common/string.h>
+
+#include <aws/testing/aws_test_harness.h>
+
+static uint64_t s_hash_string_ptr(const void *item) {
```

Comment:
- check exists false -> insert -> exists true -> remove -> exists false
- check that everything works correctly after underlying mem for both structures is reallocated?

Reply:
I create the structure with 1 element as initial size. So, when 3 elements inserted, both of the underlying mem has been reallocated.